<?xml version="1.0"?>
<launch>
  	<arg name="camera_name" default="cam0" />
  	<!-- set camera fps to -->
  	<arg name="set_camera_fps" default="20" />
  	<!-- frame_id for the camera -->
  	<arg name="frame_id" default="$(arg camera_name)" />
  	<!-- By default, calibrations are stored to file://${ROS_HOME}/camera_info/${NAME}.yaml
  	To use your own fill this arg with the corresponding url, e.g.:
  	"file:///$(find your_camera_package)/config/your_camera.yaml" -->
    <!-- <arg name="camera_info_url" default="file:///$(find offboard)/config/calibration.yaml" /> -->
    <arg name="camera_info_url" default="" />
	<!-- visualize on an image_view window the stream generated -->
	<arg name="visualize" value="false" />

   	<!-- images will be published at /camera_name/image with the image transports plugins (e.g.: compressed) installed -->
   	<group ns="$(arg camera_name)">
	    <node pkg="daheng_ros" type="daheng" name="$(arg camera_name)_stream" output="screen"> 
	    	<remap from="camera" to="image_raw" />
	    	<param name="camera_name" type="string" value="$(arg camera_name)" />
	        <param name="set_camera_fps" type="double" value="$(arg set_camera_fps)" />
	        <param name="frame_id" type="string" value="$(arg frame_id)" />
	        <param name="camera_info_url" type="string" value="$(arg camera_info_url)" />
	    </node>

	    <node if="$(arg visualize)" name="$(arg camera_name)_image_view" pkg="image_view" type="image_view">
	    	<remap from="image" to="image_raw" />
	    </node>
	</group>

</launch>
